src/services/LongTermExerciseService.ts

```diff
@@ -0,0 +1,117 @@
+  addDays(wordNodeCard.inThisSectionSince, daysToStayInASection[wordNodeCard.section]) <= new Date() ||
+  wordNodeCard.section === 0
+
+const getNumberOfWordCardsNeedingRepetition = (): Promise<number> =>
+  getWordNodeCards().then(info => info.filter(item => needsWordNodeCardRepetition(item)).length)
+
+export const getNumberOfWordCardsNeedingRepetitionWithUpperBound = async (): Promise<number> =>
+  Math.min(await getNumberOfWordCardsNeedingRepetition(), MAX_WORD_NODE_CARDS_FOR_ONE_EXERCISE)
+
+export const getNeedsRepetitionScore = (wordNodeCard: WordNodeCard): number => {
+  const daysSinceRepetitionIsNeeded = millisecondsToDays(
+    new Date().valueOf() -
+      addDays(wordNodeCard.inThisSectionSince, daysToStayInASection[wordNodeCard.section]).valueOf()
+  )
+  const daysToStayInThisSectionWithDivideByZeroProtection = Math.max(daysToStayInASection[wordNodeCard.section], 1)
+  return Math.round(daysSinceRepetitionIsNeeded / daysToStayInThisSectionWithDivideByZeroProtection)
+}
+
+export const getWordsForNextRepetition = (): Promise<WordNodeCard[]> =>
+  getWordNodeCards().then(info => {
+    const wordsToRepeat = info.filter(item => needsWordNodeCardRepetition(item))
+    if (wordsToRepeat.length <= MAX_WORD_NODE_CARDS_FOR_ONE_EXERCISE) {
+      return wordsToRepeat
+    }
+    return wordsToRepeat
+      .map(item => ({
+        ...item,
+        needsRepetitionScore: getNeedsRepetitionScore(item),
+      }))
+      .sort((word1, word2) => word2.needsRepetitionScore - word1.needsRepetitionScore)
+      .slice(0, MAX_WORD_NODE_CARDS_FOR_ONE_EXERCISE)
+      .map(({ needsRepetitionScore, ...rest }) => rest)
+  })
+
+export const getNumberOfWordsInEachSection = async (): Promise<number[]> => {
+  const result = Array(sections.length).fill(0)
+  await getWordNodeCards().then(wordNodeCards =>
+    wordNodeCards.forEach(item => {
+      result[item.section] += 1
+    })
+  )
+  return result
+}
+
+export const addWordsToFirstSection = async (words: VocabularyItem[]): Promise<void> => {
+  const wordNodeCards: WordNodeCard[] = await getWordNodeCards()
+  words.forEach(word => {
+    if (wordNodeCards.filter(item => JSON.stringify(item.word) === JSON.stringify(word)).length === 0) {
+      wordNodeCards.push({
+        word,
+        section: 0,
+        inThisSectionSince: new Date(),
+      })
+    }
+  })
+  return saveWordNodeCards(wordNodeCards)
+}
+
+export const addWordToFirstSection = async (word: VocabularyItem): Promise<void> => addWordsToFirstSection([word])
+
+const moveWordToSection = async (wordNodeCard: WordNodeCard, targetSection: number): Promise<void> => {
+  const wordNodeCards: WordNodeCard[] = await getWordNodeCards()
+  const wordNodeCardToMove = wordNodeCards.find(item => JSON.stringify(item) === JSON.stringify(wordNodeCard))
+  if (wordNodeCardToMove) {
+    wordNodeCardToMove.section = sections[Math.min(Math.max(0, targetSection), sections.length - 1)]
```

Comment:
can you explain why this is needed to get the correct section. I didn't get it yet tbh :/
i would expect `wordNodeCardToMove.section = sections[targetSection]` should be fine